eng/scripts/Language-Settings.ps1

```diff
@@ -25,6 +30,81 @@ function Get-javascript-EmitterAdditionalOptions([string]$projectDirectory) {
   return "--option @azure-tools/typespec-ts.emitter-output-dir=$projectDirectory/"
 }
 
+function Get-javascript-AdditionalValidationPackagesFromPackageSet {
+  param(
+    [Parameter(Mandatory=$true)]
+    $LocatedPackages,
+    [Parameter(Mandatory=$true)]
+    $diffObj,
+    [Parameter(Mandatory=$true)]
+    $AllPkgProps
+  )
+  $additionalValidationPackages = @()
+
+  function isOther($fileName) {
+    $startsWithPrefixes = @(".config", ".devcontainer", ".github", ".scripts", ".vscode", "common", "design", "documentation", "eng", "samples")
+
+    $startsWith = $false
+    foreach ($prefix in $startsWithPrefixes) {
+      if ($fileName.StartsWith($prefix)) {
+        $startsWith = $true
+      }
+    }
+
+    return $startsWith
+  }
+
+  $changedServices = @()
+  foreach($file in $diffObj.ChangedFiles) {
+    $pathComponents = $file -split "/"
+    # handle changes only in sdk/<service>/<file>/<extension>
+    if ($pathComponents.Length -eq 3 -and $pathComponents[0] -eq "sdk") {
+      $changedServices += $pathComponents[1]
+    }
+
+    # handle any changes under sdk/<file>.<extension>
+    if ($pathComponents.Length -eq 2 -and $pathComponents[0] -eq "sdk") {
+      $changedServices += "template"
+    }
+  }
+
+  $engChanged = $diffObj.ChangedFiles | Where-Object { $_.StartsWith("eng")}
+  $othersChanged = $diffObj.ChangedFiles | Where-Object { isOther($_) }
+  $changedServices = $changedServices | Get-Unique
+
+  if ($engChanged -or $othersChanged) {
+    $additionalPackages = $ReducedDependencyLookup["core"] | ForEach-Object { $me=$_; $AllPkgProps | Where-Object { $_.Name -eq $me } | Select-Object -First 1 }
+    $additionalValidationPackages += $additionalPackages
+  }
+
+  foreach ($changedService in $changedServices) {
+    if ($ReducedDependencyLookup.ContainsKey($changedService)) {
+      $additionalPackages = $ReducedDependencyLookup[$changedService] | ForEach-Object { $me=$_; $AllPkgProps | Where-Object { $_.Name -eq $me } | Select-Object -First 1 }
+      $additionalValidationPackages += $additionalPackages
+    }
+    else {
+      $additionalPackages = $AllPkgProps | Where-Object { $_.ServiceDirectory -eq $changedService }
+      $additionalValidationPackages += $additionalPackages
+    }
+  }
+
+  $uniqueResultSet = @()
+  foreach ($pkg in $additionalValidationPackages) {
+    if ($uniqueResultSet -notcontains $pkg -and $LocatedPackages -notcontains $pkg) {
+      $pkg.IncludedForValidation = $true
```

Comment:
This is unused?

Reply:
?? Will confirm.

Reply:
Sorry, I'm a dummy. Marking this as resolved.